src/common/snippets/src/lowered/expression.cpp

```diff
@@ -96,13 +95,17 @@ void Expression::set_reg_info(const RegInfo& rinfo) {
     }
 }
 
+ void Expression::validate_attributes() const {
+    OPENVINO_ASSERT(m_source_node != nullptr,
+                "The expression has null source node");
+ }
+
 void Expression::validate() const {
+    validate_attributes();
     OPENVINO_ASSERT(m_input_port_descriptors.size() == m_input_port_connectors.size(),
                     "The count of input ports and input port connectors must be equal");
     OPENVINO_ASSERT(m_output_port_descriptors.size() == m_output_port_connectors.size(),
                     "The count of output ports and output port connectors must be equal");
-    OPENVINO_ASSERT(m_source_node != nullptr,
-                    "The expression has null source node");
```

Comment:
Actually port descriptors/connectors are Expression's attribute as `m_source_node`. Do we need to have 2 separate virtual methods `validate` and `validate_attributes`? Maybe it is better to combine them into one? I suppose this would be enough: in `BufferExpression`, we would just call `Expression::validate`, and then check source node type. What do you think?